### PR TITLE
Fix offset between data and times and durations

### PR DESCRIPTION
**Pull request by samaloney**
I'm not sure how useful this really is as what we really need to know is the requested time bin size. As long as the time bins are `>=` the requested time bin size should be safe of shift otherwise not.

### CODE REVIEW

**Comment by samaloney** [comment]
Discussed today and decided we should do where we know we can (most spectrogram and flare data) and add a flag or comment when we can't but need to.

File: setup.cfg [hunk 1]
```diff
@@ -12,7 +12,7 @@ long_description = file: README.rst
 include_package_data = True
 zip_safe = False
 packages = find:
-python_requires = >=3.7
+python_requires = >=3.8
```
Comment (samaloney) [inline setup.cfg:15]: I think this is causing the issues with the tests

File: stixcore/products/product.py [hunk 2]
```diff
@@ -38,6 +39,7 @@
     32: np.uint32,
     64: np.uint64
 }
+DURATION_FIX_DATE = datetime(2021, 12, 9, 13, 0, tzinfo=timezone.utc)
```
Comment (nicHoch) [inline stixcore/products/product.py:43]: still needed?

File: stixcore/products/product.py [hunk 3]
```diff
@@ -706,6 +708,36 @@ def from_level0(cls, l0product, parent=''):
...
+                    correct_triggers = l1.data['triggers'][1:, :]
+                    correct_trigger_errs = l1.data['triggers_err'][1:, :]
+                    correct_times = l1.data['timedel'][1:]
+                    correct_durations = l1.data['timedel'][1:]
+                    correct_rcr = l1.data['rcr'][1:]
+                    l1.data = l1.data[0:-1]
+                    l1.data['counts'] = correct_counts
+                    l1.data['counts_err'] = correct_count_errs
+                    l1.data['triggers'] = correct_triggers
+                    l1.data['triggers_err'] = correct_trigger_errs
```
Comment (nicHoch) [inline stixcore/products/product.py:736]: this is a nice and compact fix i was afraid it would have to be implemented on many more places.

**Review by nicHoch — APPROVED** [review]
good

File: stixcore/products/product.py (hunk 2 above)
Comment (samaloney) [reply stixcore/products/product.py:43]: yea I think my plan was to replace the datetime below I will do that now so would only be created once and then referenced

**Comment by samaloney** [comment]
I think the only aspect missing is adding a comment when we know it should be fixed but we can't but this will be a fair but of work as have to add comments to the products and IO

**Comment by nicHoch** [comment]
there is a "header" property on products if it was open from fits
i am not sure at what level 
but you could test if the property is around  and if so you can add the comment fairly easy

**Comment by samaloney** [comment]
I don't quiet follow, what do you meant add the comment?

File: stixcore/products/product.py [hunk 4]
```diff
@@ -130,18 +135,21 @@ def __call__(self, *args, **kwargs):
...
                 file_path = Path(args[0])
-                header = fits.getheader(file_path)
+                pri_header = fits.getheader(file_path)
+
+                service_type = int(pri_header.get('stype')) if 'stype' in pri_header else 0
+                service_subtype = int(pri_header.get('sstype')) if 'sstype' in pri_header else 0
+                parent = pri_header.get('parent').split(';') if 'parent' in pri_header else ''
+                raw = pri_header.get('raw_file').split(';') if 'raw_file' in pri_header else ''
+                comment = [c for c in pri_header.get('comment', [])]
+                history = [h for h in pri_header.get('history', [])]
```
Comment (nicHoch) [inline stixcore/products/product.py:145]: ahh it is also reading in 
this is nice

File: stixcore/products/product.py [hunk 5]
```diff
@@ -704,7 +716,43 @@ def from_level0(cls, l0product, parent=''):
...
+                    l1.data['counts_err'] = correct_count_errs
+                    l1.data['triggers'] = correct_triggers
+                    l1.data['triggers_err'] = correct_trigger_errs
+                    l1.data['time'] = correct_times
+                    l1.data['timedel'] = correct_durations
+                    l1.data['rcr'] = correct_rcr
+                    l1.history.append('Time and count arrays were shifted to fix offset')
+                else:
+                    l1.comment.append('Time and count arrays offset not fixed'
+                                      ' as possibly summed on board')
```
Comment (nicHoch) [inline stixcore/products/product.py:755]: i know about that but somehow i think other users might not could we add a link to some kind of online documentation / git issue/ dpdd that will provide more background
?

**Review by nicHoch — APPROVED** [review]
i see now why you complained about the inheritance

File: stixcore/products/product.py (hunk 5 above)
Comment (samaloney) [reply stixcore/products/product.py:755]: Yea I think we need a read me with the data release summarising the data and know issues.

**Comment by samaloney** [comment]
Closes #291

**Pull request merged**